packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -146,56 +144,117 @@ function ManualGridVisualizer( { gridClientId, gridInfo } ) {
 		return rects;
 	}, [ gridItems ] );
 
-	return range( 1, gridInfo.numRows ).map( ( row ) =>
-		range( 1, gridInfo.numColumns ).map( ( column ) => {
-			const isCellOccupied = occupiedRects.some( ( rect ) =>
-				rect.contains( column, row )
-			);
-			const isHighlighted =
-				highlightedRect?.contains( column, row ) ?? false;
-			return (
-				<GridVisualizerCell
-					key={ `${ row }-${ column }` }
-					color={ gridInfo.currentColor }
-					className={ isHighlighted && 'is-highlighted' }
-				>
-					{ isCellOccupied ? (
-						<GridVisualizerDropZone
-							column={ column }
-							row={ row }
-							gridClientId={ gridClientId }
-							gridInfo={ gridInfo }
-							setHighlightedRect={ setHighlightedRect }
-						/>
-					) : (
-						<GridVisualizerAppender
-							column={ column }
-							row={ row }
-							gridClientId={ gridClientId }
-							gridInfo={ gridInfo }
-							setHighlightedRect={ setHighlightedRect }
-						/>
-					) }
-				</GridVisualizerCell>
-			);
-		} )
+	return (
+		<BlockPopoverCover
+			__unstablePopoverSlot="__unstable-block-tools-after"
+			className={ clsx( 'block-editor-grid-visualizer', {
+				'is-dropping-allowed': isDroppingAllowed,
+			} ) }
+			clientId={ gridClientId }
+		>
+			<div
+				ref={ ref }
+				className="block-editor-grid-visualizer__grid"
+				style={ gridInfo.style }
+			>
+				{ range( 1, gridInfo.numRows ).map( ( row ) =>
+					range( 1, gridInfo.numColumns ).map( ( column ) => {
+						const isCellOccupied = occupiedRects.some( ( rect ) =>
+							rect.contains( column, row )
+						);
+						const isHighlighted =
+							highlightedRect?.contains( column, row ) ?? false;
+
+						return (
+							<div
+								key={ `${ row }-${ column }` }
+								className={ clsx(
+									'block-editor-grid-visualizer__cell',
+									{
+										'is-highlighted': isHighlighted,
+									}
+								) }
+							>
+								{ isCellOccupied ? (
+									<GridVisualizerDropZone
+										column={ column }
+										row={ row }
+										gridClientId={ gridClientId }
+										gridInfo={ gridInfo }
+										setHighlightedRect={
+											setHighlightedRect
+										}
+									/>
+								) : (
+									<GridVisualizerAppender
+										column={ column }
+										row={ row }
+										gridClientId={ gridClientId }
+										gridInfo={ gridInfo }
+										setHighlightedRect={
+											setHighlightedRect
+										}
+									/>
+								) }
+							</div>
+						);
+					} )
+				) }
+			</div>
+		</BlockPopoverCover>
 	);
-}
+} );
 
-function GridVisualizerCell( { color, children, className } ) {
```

Comment:
I decided to get rid of `GridVisualizerCell` as it wasn't doing very much, especially in `GridPopover` where it's just a `<div>`.

In `GridPopunder` it's a `<div>` with a some styles, so I think it still doesn't need its own component.